pages/docs/reference/whatsnew1420.md

```diff
@@ -0,0 +1,348 @@
+- [Escape analysis](#escape-analysis)
+- [Opt-in wrapping of Objective-C exceptions](#opt-in-wrapping-of-objective-c-exceptions)
+- [Support for Xcode 12 libraries](#support-for-xcode-12-libraries)
+- [Performance improvements and bug fixes](#performance-improvements-and-bug-fixes)
+
+[**Kotlin Multiplatform**](#kotlin-multiplatform)
+- [Updated structure of multiplatform library publications](#updated-structure-of-multiplatform-library-publications)
+
+[**CocoaPods support**](#cocoapods-support)
+- [Improved task execution](#improved-task-execution)
+- [Extended DSL](#extended-dsl)
+- [Updated integration with Xcode](#updated-integration-with-xcode)
+
+[**Standard library**](#standard-library)
+- [Extensions for java.nio.file.Path](#extensions-for-javaniofilepath)
+- [Improved String.replace function performance](#improved-stringreplace-function-performance)
+
+[**Kotlin Android Extensions**](#kotlin-android-extensions)
+- [Deprecation of synthetic views](#deprecation-of-synthetic-views)
+- [New plugin for Parcelable implementation generator](#new-plugin-for-parcelable-implementation-generator)
+
+## Kotlin/JVM
+
+### Java 15 target
+
+Now Java 15 is available as a Kotlin/JVM target.
+
+### invokedynamic string concatenation
+
+> `invokedynamic` string concatenation is [experimental](evolution/components-stability.html) in Kotlin 1.4.20. It may change incompatibly in future versions.
+{:.note}
+
+Kotlin 1.4.20 can compile string concatenations into [dynamic invocations](https://docs.oracle.com/javase/7/docs/technotes/guides/vm/multiple-language-support.html#invokedynamic)
+on JVM 9+ targets, therefore improving the performance.
+
+Currently, this feature is experimental and covers the following cases:
+- `String.plus` in the operator (`a + b`), explicit (`a.plus(b)`), and reference (`(a::plus)(b)`) form.
+- `toString` on inline and data classes.
+- string templates except for ones with a single non-constant argument (see [KT-42457](https://youtrack.jetbrains.com/issue/KT-42457)).
+
+To enable `invokedynamic` string concatenation, add the `-Xstring-concat` compiler option with one of the following values:
+- `indy-with-constants` to perform `invokedynamic` concatenation on strings with [StringConcatFactory.makeConcatWithConstants()](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/StringConcatFactory.html#makeConcatWithConstants-java.lang.invoke.MethodHandles.Lookup-java.lang.String-java.lang.invoke.MethodType-java.lang.String-java.lang.Object...-)
+    (this is planned to be the default for JVM 9+ targets in 1.5).
+- `indy` to perform `invokedynamic` concatenation on strings with [StringConcatFactory.makeConcat()](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/StringConcatFactory.html#makeConcat-java.lang.invoke.MethodHandles.Lookup-java.lang.String-java.lang.invoke.MethodType-).
+- `inline` to switch back to the classic concatenation via `StringBuilder.append()`.
+
+[**Back to top**](#)
+
+## Kotlin/JS
```

Comment:
I would add a summary for each section that includes multiple elements. It simplifies scanning and finding what you need without reading all content. Also it makes sense to summarize purpose of changes and improvements in this area.
Applies to all similar cases in the doc.